src/com/facebook/buck/go/GoBuckConfig.java

```diff
@@ -109,7 +110,11 @@ Path getDefaultPackageName(BuildTarget target) {
       public Tool get() {
         Optional<Path> toolPath = delegate.getPath("go", configName);
         if (!toolPath.isPresent()) {
-          toolPath = Optional.of(goToolDirSupplier.get().resolve(toolName));
+          String exeName = toolName;
+          if (delegate.getPlatform() == Platform.WINDOWS) {
+            exeName = toolName + ".exe";
+          }
```

Comment:
I was looking at that, but I *think* the "fast path" (https://github.com/facebook/buck/blob/master/src/com/facebook/buck/io/ExecutableFinder.java#L121) makes it a bit weird. If you create a file called "compile" in the root of your project, it will prefer that even with an explicit directory passed in. Is that intentional?